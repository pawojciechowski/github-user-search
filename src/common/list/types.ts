export interface ListItem {
  id: string;
  text: string;
  href?: string;
}
