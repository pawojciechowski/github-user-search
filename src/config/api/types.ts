// TODO: Verify types

export interface GithubUserResponse {
  login: string;
  name: string | null;
  bio: string | null;
  avatar_url: string;
}

export interface GithubReposResponseRepository {
  id: number;
  name: string;
  html_url: string;
  stargazers_count: number;
}

export interface GithubReposResponse extends Array<GithubReposResponseRepository> {}
