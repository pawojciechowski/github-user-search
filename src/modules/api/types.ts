export type APIUrl = string;

export interface IAPIService {
  get<T>(url: APIUrl, params?: { [key: string]: string }): Promise<T>;
}

export interface APIServiceOptions {
  headers?: { [key: string]: string };
  mockResponse?: { [key: string]: any };
  logRequests?: boolean;
}
