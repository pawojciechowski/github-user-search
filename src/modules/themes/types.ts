import { themeNames } from './themes';

export type Color = string;
export type FontSize = string;
export type FontWeight = string;
export type FontFamily = string;
export type LineHeight = string;
export type LetterSpacing = string;

interface FontSettings {
  fontFamily?: FontFamily,
  fontSize?: FontSize,
  fontWeight?: FontWeight,
  lineHeight?: LineHeight,
  letterSpacing?: LetterSpacing,
  color?: Color
}

interface InputStyles {
  placeholder?: {
    text: FontSettings
  },
  background: Color,
  text: FontSettings
}

interface ButtonStyles {
  background: Color,
  text: FontSettings
}

interface TypographySettings {
  base: FontSettings,
  title: FontSettings
}

export interface Theme {
  background: Color,
  input: InputStyles,
  button: ButtonStyles,
  spacer: number,
  typography: TypographySettings,
}

export type ThemeName = typeof themeNames[number];
