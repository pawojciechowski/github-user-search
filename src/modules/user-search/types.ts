export interface GithubUserResponse {
  login: string,
  name: string,
  bio: string | null,
  avatar_url: string
}

export interface GithubReposResponseRepo {
  id: number,
  name: string,
  html_url: string,
  stargazers_count: number
}

export interface GithubReposResponse extends Array<GithubReposResponseRepo> {}
