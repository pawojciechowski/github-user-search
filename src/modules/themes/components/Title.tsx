import styled from 'styled-components';
import { Theme } from '../types';

const Title = styled.h1`
  font-size: ${({ theme }: { theme: Theme}) => theme.typography.title.fontSize};
  font-weight: ${({ theme }: { theme: Theme}) => theme.typography.title.fontWeight};
  line-height: ${({ theme }: { theme: Theme}) => theme.typography.title.lineHeight};
  letter-spacing: ${({ theme }: { theme: Theme}) => theme.typography.title.letterSpacing};
  color: ${({ theme }: { theme: Theme}) => theme.typography.title.color};
  margin: 0;
  padding: 0;
`;

export default Title;
