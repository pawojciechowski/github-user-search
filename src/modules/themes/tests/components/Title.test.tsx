import React from 'react';
import { render } from '@testing-library/react';
import Title from '../../components/Title';
import { Theme } from '../../types';

const testTheme = {
  typography: {
    title: {
      fontSize: 'test',
      fontWeight: 'test',
      lineHeight: 'test',
      letterSpacing: 'test',
      color: 'test',
    },
  },
  spacer: 1,
} as Theme;

describe('Title', () => {
  it('matches snapshot', () => {
    const { container } = render(<Title theme={testTheme} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
