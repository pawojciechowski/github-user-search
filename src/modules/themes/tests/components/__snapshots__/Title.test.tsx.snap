// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> matches snapshot 1`] = `
.c0 {
  font-size: test;
  font-weight: test;
  line-height: test;
  -webkit-letter-spacing: test;
  -moz-letter-spacing: test;
  -ms-letter-spacing: test;
  letter-spacing: test;
  color: test;
  margin-top: 0;
  margin-bottom: 1px;
  padding: 0;
}

<h1
  class="c0"
/>
`;
